src/main/java/org/jabref/gui/autosaveandbackup/BackupManager.java

```diff
@@ -1,384 +1,705 @@
     }
 
     /**
-     * Restores the backup file by copying and overwriting the original one.
+     * Performs the backup by checking for changes and committing them to the Git repository.
      *
-     * @param originalPath Path to the file which should be equalized to the backup file.
+     * @param backupDir the backup directory
+     * @param dbfile the database file
+     * @throws IOException     if an I/O error occurs
+     * @throws GitAPIException if a Git API error occurs
      */
-    public static void restoreBackup(Path originalPath, Path backupDir) {
-        Optional<Path> backupPath = getLatestBackupPath(originalPath, backupDir);
-        if (backupPath.isEmpty()) {
-            LOGGER.error("There is no backup file");
+
+    protected void performBackup(Path dbfile, Path backupDir) throws IOException, GitAPIException {
+
+        boolean needsCommit = backupGitDiffers(dbfile, backupDir);
+
+        if (!needsBackup && !needsCommit) {
+            LOGGER.info("No changes detected, beacuse needsBackup is :{} and needsCommit is :{}", needsBackup, needsCommit);
             return;
         }
-        try {
-            Files.copy(backupPath.get(), originalPath, StandardCopyOption.REPLACE_EXISTING);
-        } catch (IOException e) {
-            LOGGER.error("Error while restoring the backup file.", e);
-        }
+
+        if (needsBackup) {
+            LOGGER.info("Backup needed, because needsBackup is :{}", needsBackup);
+        } else {
+        LOGGER.info("Backup needed, because needsCommit is :{}", needsCommit);
     }
 
-    Optional<Path> determineBackupPathForNewBackup(Path backupDir) {
-        return bibDatabaseContext.getDatabasePath().map(path -> BackupManager.getBackupPathForNewBackup(path, backupDir));
+        // Stage the file for commit
+        git.add().addFilepattern(".").call();
+        LOGGER.info("Staged changes for backup in directory: {}", backupDir);
+
+        // Commit the staged changes
+        RevCommit commit = git.commit()
+                              .setMessage("Backup at " + Instant.now().toString())
+                              .call();
+        LOGGER.info("Backup committed in :{} with commit ID: {} for the file : {}", backupDir, commit.getName(), bibDatabaseContext.getDatabasePath().orElseThrow());
+    }
+
+    public synchronized void listen(BibDatabaseContextChangedEvent event) {
+        if (!event.isFilteredOut()) {
+            LOGGER.info("Change detected/LISTENED in file: {}", bibDatabaseContext.getDatabasePath().orElseThrow());
+            this.needsBackup = true;
+        }
     }
 
     /**
-     * This method is called as soon as the scheduler says: "Do the backup"
+     * Restores the backup from the specified commit.
      *
-     * <em>SIDE EFFECT: Deletes oldest backup file</em>
+     * @param backupDir the backup directory
+     * @param objectId the commit ID to restore from
+     */
+
```

Comment:
JavaDoc and method begin should be kept together - please remove all empty lines between JavaDoc and method.